content/docs/guides/author_develop/pwa-amp/amp-as-pwa.md

```diff
@@ -0,0 +1,110 @@
+{% endcall %}
+
+## Add a Web App Manifest
+
+Adding a [Web App Manifest](https://developers.google.com/web/fundamentals/engage-and-retain/web-app-manifest/) to your AMP pages ensures that users can install your site to their devices' homescreen. There's nothing special about Web App manifests in AMP.
+
+First, create the manifest:
+
+[sourcecode:json]
+{
+  "short_name": "ABE",
+  "name": "AMPByExample",
+  "icons": [
+    {
+      "src": "launcher-icon-1x.png",
+      "type": "image/png",
+      "sizes": "48x48"
+    },
+    {
+      "src": "launcher-icon-2x.png",
+      "type": "image/png",
+      "sizes": "96x96"
+    },
+    {
+      "src": "launcher-icon-4x.png",
+      "type": "image/png",
+      "sizes": "192x192"
+    }
+  ],
+  "start_url": "index.html?launcher=true"
+}
+[/sourcecode]
+
+Then link to it from the `<head>` of your AMP page:
+
+[sourcecode:html]
+<link rel="manifest" href="/manifest.json">
+[/sourcecode]
+
+{% call callout('Tip', type='success') %}
+Learn more about the [Web App Manifest at WebFundamentals](https://developers.google.com/web/fundamentals/engage-and-retain/web-app-manifest/).
+{% endcall %}
+
+## Install a Service Worker to enable offline access
+
+A Service Worker is a client-side proxy that sits between your page and your server, and can be used to build fantastic offline experiences, fast-loading app shell scenarios and the sending of push notifications.
+
+{% call callout('Note', type='note') %}
+If the concept of Service Workers is new to you, read the [introduction at WebFundamentals](https://developers.google.com/web/fundamentals/getting-started/primers/service-workers).
+{% endcall %}
+
+As Service Worker needs to be registered on a given page, or the browser won't find or run it. By default, this is done with the help of a [little bit of JavaScript](https://developers.google.com/web/fundamentals/instant-and-offline/service-worker/registration). On AMP Pages, you use the [`<amp-install-serviceworker>`](/docs/reference/components/dynamic/amp-install-serviceworker) component to achieve the same, like so:
+
+[sourcecode:html]
+<amp-install-serviceworker
+      src="https://www.your-domain.com/serviceworker.js"
+      layout="nodisplay">
+</amp-install-serviceworker>
+[/sourcecode]
+
+If the user navigates to your AMP pages on your origin (as opposed to the first click, which is usually served from an AMP Cache), the Service Worker will take over and can do a [myriad of cool things](https://developers.google.com/web/fundamentals/instant-and-offline/offline-ux).
+
+## Extend your AMP Pages via Service Worker
+
+You can use the above technique to enable offline access to your AMP website, as well as extend your pages **as soon as they’re served from the origin**. That's because you can modify the response via the Service Worker’s `fetch` event, and return as a response whatever you want:
```

Comment:
Consider rewording "return as a response whatever you want:" to "return any response you want:"